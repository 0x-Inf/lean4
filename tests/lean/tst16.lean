variable f : Type -> Bool
print forall a b : Type, (f a) = (f b)
variable g : Bool -> Bool -> Bool
print forall (a b : Type) (c : Bool), g c ((f a) = (f b))
print exists (a b : Type) (c : Bool), g c ((f a) = (f b))
print forall (a b : Type) (c : Bool), (g c (f a) = (f b)) ⇒ (f a)
check forall (a b : Type) (c : Bool), g c ((f a) = (f b))
print ∀ (a b : Type) (c : Bool), g c ((f a) = (f b))
print ∀ a b : Type, (f a) = (f b)
print ∃ a b : Type, (f a) = (f b) ∧ (f a)
print ∃ a b : Type, (f a) = (f b) ∨ (f b)
variable a : Bool
print (f a) ∨ (f a)
print (f a) = a ∨ (f a)
print (f a) = a ∧ (f a)
