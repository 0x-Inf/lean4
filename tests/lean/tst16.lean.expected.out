  Assumed: f
[33mâˆ€[0m a b : [36mType[0m, (f a) = (f b)
  Assumed: g
[33mâˆ€[0m (a b : [36mType[0m) (c : Bool), g c ((f a) = (f b))
[33mâˆƒ[0m (a b : [36mType[0m) (c : Bool), g c ((f a) = (f b))
[33mâˆ€[0m (a b : [36mType[0m) (c : Bool), (g c (f a)) = (f b) â‡’ (f a)
Bool
[33mâˆ€[0m (a b : [36mType[0m) (c : Bool), g c ((f a) = (f b))
[33mâˆ€[0m a b : [36mType[0m, (f a) = (f b)
[33mâˆƒ[0m a b : [36mType[0m, (f a) = (f b) âˆ§ (f a)
[33mâˆƒ[0m a b : [36mType[0m, (f a) = (f b) âˆ¨ (f b)
  Assumed: a
(f a) âˆ¨ (f a)
(f a) = a âˆ¨ (f a)
(f a) = a âˆ§ (f a)
